src/EventEdition/EventDialog.vala

```diff
@@ -214,9 +214,14 @@ public class EventDialog : Granite.Dialog {
         }
 
         private void remove_event () {
-            var calmodel = Calendar.EventStore.get_default ();
-            calmodel.remove_event (original_source, ecal, mod_type);
-            this.destroy ();
+            assert (original_source != null);
+            var delete_dialog = new Calendar.DeleteEventDialog (original_source, ecal, mod_type) {
+                transient_for = this
+            };
+            var response = delete_dialog.run_dialog ();
+            if (response == Gtk.ResponseType.YES) {
```

Comment:
It would be usual for the `delete_dialog` to be destroyed explicitly after getting the response, rather than trusting it will be destroyed when it goes out of scope.